numpy/f2py/tests/test_crackfortran.py

```diff
@@ -33,3 +36,50 @@ def test_module(self):
         self.module.subc([w, k])
         assert_array_equal(k, w + 1)
         assert self.module.t0(23) == b'2'
+
+class TestPublicPrivate():
+    def test_defaultPrivate(self):
+        with tempfile.NamedTemporaryFile() as ff:
+            ff.write(textwrap.dedent("""\
+            module foo
+              private
+              integer :: a
+              public :: setA
+            contains
+              subroutine setA(v)
+                integer, intent(in) :: v
+                a = v
+              end subroutine setA
+            end module foo
+            """).encode())
+            ff.flush()
+            mod = crackfortran.crackfortran([ff.name, ])
+            assert len(mod) == 1
+            mod = mod[0]
+            assert 'private' in mod['vars']['a']['attrspec']
+            assert 'public' not in mod['vars']['a']['attrspec']
+            assert 'private' not in mod['vars']['seta']['attrspec']
+            assert 'public' in mod['vars']['seta']['attrspec']
```

Comment:
```suggestion
        mod = crackfortran.crackfortran([str(f_path)])
        assert len(mod) == 1
        mod = mod[0]
        assert 'private' in mod['vars']['a']['attrspec']
        assert 'public' not in mod['vars']['a']['attrspec']
        assert 'private' not in mod['vars']['seta']['attrspec']
        assert 'public' in mod['vars']['seta']['attrspec']
```

Reply:
Code moved out of the `with` as suggested.